test/acceptance_tests/acceptancetest_high_cpu.py

```diff
@@ -1,36 +1,84 @@
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import time
+import sys
 import psutil
 
-OBSERVATION_FREQUENCY = .5
+OBSERVATION_FREQUENCY_HZ = .5
+OBSERVATION_DURATION_SEC = 60
 PSEN_SCAN_PROCESS_NAME = "psen_scan_node"
 NAME = 'name'
-TRESHHOLD_PRECENT = 30
+THRESHOLD_PRECENT = 80
+RETURN_CODE_SUCCESS = 0
+RETURN_CODE_FAILURE = 10
+RETURN_CODE_ERROR = 1
 
+# ASCII codes for printing colored text to console
 GREEN = '\033[92m'
 YELLOW = '\033[93m'
 RED = '\033[91m'
 NC = '\033[0m'
 
 
-def get_psenscan_process():  # type: () -> Union[None, psutil.Process]
-    """Return the handle of the psenscan process if it is currently running,
+def get_psen_scan_process():  # type: () -> psutil.Process
+    """Return the handle of the psen_scan process if it is currently running,
     None otherwise."""
     for proc in psutil.process_iter([NAME]):
         if proc.info[NAME] == PSEN_SCAN_PROCESS_NAME:
             return proc
-    return None
-
-
-while True:
-    process = get_psenscan_process()
-    if process is not None:
-        utilization = process.cpu_percent()
-        color = GREEN if utilization < TRESHHOLD_PRECENT else RED
-        print(color + "CPU utilization of >" + PSEN_SCAN_PROCESS_NAME + "<: " +
-              str(utilization) + "%" + NC)
-    else:
-        print(YELLOW + "psen_scan process seems not to be running." + NC)
-    time.sleep(1. / OBSERVATION_FREQUENCY)
+    raise RuntimeError("Process %s not found." % PSEN_SCAN_PROCESS_NAME)
+
+
+if __name__ == "__main__":
+    process = None
+    while process is None:
+        print("Waiting for %s process to be running." %
+              PSEN_SCAN_PROCESS_NAME)
+        try:
+            process = get_psen_scan_process()
+        except RuntimeError:
+            process = None
+        time.sleep(1. / OBSERVATION_FREQUENCY_HZ)
+    print(YELLOW +
+          "Process is running. Will observe its CPU usage now for %d seconds" %
+          OBSERVATION_DURATION_SEC + NC)
+    start_time = time.time()
+    while time.time() - start_time < OBSERVATION_DURATION_SEC:
+        try:
+            process = get_psen_scan_process()
+        except RuntimeError:
+            process = None
+        if process is None:
+            print(RED + "Process not running any more. Aborting Test" + NC)
+            sys.exit(RETURN_CODE_FAILURE)
+        else:
```

Comment:
This could reduce indentation a little.

```suggestion
            print(RED + "Process not running any more. Aborting Test" + NC)
            sys.exit(RETURN_CODE_FAILURE)
```